samples/dnn/DaSiamRPN_visual_tracker.py

```diff
@@ -0,0 +1,330 @@
+
+    delta[0, :] = delta[0, :] * p.anchor[:, 2] + p.anchor[:, 0]
+    delta[1, :] = delta[1, :] * p.anchor[:, 3] + p.anchor[:, 1]
+    delta[2, :] = np.exp(delta[2, :]) * p.anchor[:, 2]
+    delta[3, :] = np.exp(delta[3, :]) * p.anchor[:, 3]
+
+    def change(r):
+        return np.maximum(r, 1./r)
+
+    def sz(w, h):
+        pad = (w + h) * 0.5
+        sz2 = (w + pad) * (h + pad)
+        return np.sqrt(sz2)
+
+    def sz_wh(wh):
+        pad = (wh[0] + wh[1]) * 0.5
+        sz2 = (wh[0] + pad) * (wh[1] + pad)
+        return np.sqrt(sz2)
+
+    s_c = change(sz(delta[2, :], delta[3, :]) / (sz_wh(target_sz)))
+    r_c = change((target_sz[0] / target_sz[1]) / (delta[2, :] / delta[3, :]))
+
+    penalty = np.exp(-(r_c * s_c - 1.) * p.penalty_k)
+    pscore = penalty * score
+
+    pscore = pscore * (1 - p.window_influence) + window * p.window_influence
+    best_pscore_id = np.argmax(pscore)
+
+    target = delta[:, best_pscore_id] / scale_z
+    target_sz = target_sz / scale_z
+    lr = penalty[best_pscore_id] * score[best_pscore_id] * p.lr
+
+    res_x = target[0] + target_pos[0]
+    res_y = target[1] + target_pos[1]
+
+    res_w = target_sz[0] * (1 - lr) + target[2] * lr
+    res_h = target_sz[1] * (1 - lr) + target[3] * lr
+
+    target_pos = np.array([res_x, res_y])
+    target_sz = np.array([res_w, res_h])
+
+    return target_pos, target_sz, score[best_pscore_id]
+
+def softmax(x):
+    y = np.copy(x)
+
+    for i in range(1805):
+        e_x = np.exp(x[: , i] - np.max(x[ : , i]))
+        y[ : , i] = e_x /e_x.sum()
+
+    return y
+
+#parse paths to onnx models and to input sequence
+parser = argparse.ArgumentParser(description = "Run tracker")
+parser.add_argument("net", type = str, help = "Full path to onnx model of net")
+parser.add_argument("kernel_r1", type = str, help = "Full path to onnx model of kernel_r1")
+parser.add_argument("kernel_cls1", type = str, help = "Full path to onnx model of kernel_cls1")
+# parser.add_argument("input", type = str, help = "Full path to input sequence")
+args = parser.parse_args()
+
+net = cv.dnn.readNetFromONNX(args.net)
+kernel_r1 = cv.dnn.readNetFromONNX(args.kernel_r1)
+kernel_cls1 = cv.dnn.readNetFromONNX(args.kernel_cls1)
+
+#read source of video/image sequence
+#for now it should be folder with images named like: "0001", "0002" etc. for correct work "sorted" method
+# images = sorted(glob.glob(args.input + '*.jpg'))
+
+#choose region on image
+#region = [334.02,128.36,438.19,188.78,396.39,260.83,292.23,200.41]
+#[cx, cy, w, h] = get_axis_aligned_bbox(region)
+[cx, cy, w, h] = [365.2075, 194.595, 106.13, 96.41]
+
+#initialization of network and initial bounding box
+target_pos, target_sz = np.array([cx, cy]), np.array([w, h])
+# im = cv.imread(images[0])
+# state = SiamRPN_init(im, target_pos, target_sz, net, kernel_r1, kernel_cls1)
+toc = 0#variable for fps counter
+cap = cv.VideoCapture("/home/ilyaelizarov/trackers/DaSiamRPN/code/bag/%08d.jpg",cv.CAP_IMAGES)
```

Comment:
These absolute paths are not allowed.

Reply:
It is gonna be replaced with argparse for read the path to model.